avocado/core/varianter.py

```diff
@@ -447,115 +381,66 @@ def _skip_new_data_check(self, function, args):
-            for (index, tpl) in enumerate(self.variants):
-                if not self.debug:
-                    paths = ', '.join([x.path for x in tpl])
-                else:
-                    color = output.TERM_SUPPORT.LOWLIGHT
-                    cend = output.TERM_SUPPORT.ENDC
-                    paths = ', '.join(["%s%s@%s%s" % (_.name, color,
-                                                      getattr(_, 'yaml',
-                                                              "Unknown"),
-                                                      cend)
-                                       for _ in tpl])
-                out.append('%sVariant %s:    %s' % ('\n' if contents else '',
-                                                    index + 1, paths))
-                if contents:
-                    env = set()
-                    for node in tpl:
-                        for key, value in node.environment.iteritems():
-                            origin = node.environment_origin[key].path
-                            env.add(("%s:%s" % (origin, key), str(value)))
-                    if not env:
-                        continue
-                    fmt = '    %%-%ds => %%s' % max([len(_[0]) for _ in env])
-                    for record in sorted(env):
-                        out.append(fmt % record)
-        return "\n".join(out)
+        return "\n\n".join(self._variant_plugins.map_method("to_str", summary,
+                                                            variants,
+                                                            **kwargs))
 
     def get_number_of_tests(self, test_suite):
         """
         :return: overall number of tests * number of variants
         """
         # Currently number of tests is symmetrical
-        if self.variants:
-            no_variants = len(self.variants)
-            return len(test_suite) * no_variants
+        if self._no_variants:
+            return len(test_suite) * self._no_variants
         else:
             return len(test_suite)
 
-    def _merge_defaults(self, variant):
-        """
-        Copy the default params tree, merge the individual leaves of the
-        variant and return the combination's leaves (variant with defaults)
-        """
-        data = copy.deepcopy(self.default_params)
-        for leaf in variant:
-            data.get_node(leaf.path, True).merge(leaf)
-        data.set_environment_dirty()
-        return data.get_leaves()
-
     def itertests(self):
         """
         Yield variant-id and test params
 
         :yield (variant-id, (list of leaves, list of default paths))
         """
-        if self.variants:  # Copy template and modify it's params
-            for i, variant in enumerate(self.variants, 1):
-                yield i, (variant, self._mux_path)
+        if self._no_variants:  # Copy template and modify it's params
+            pluginss_variants = self._variant_plugins.map_method("__iter__")
```

Comment:
OK, I meant it as a shortcut to `plugins's variants` but it probably does not work like that in English...